File: ReverseString.js
/**
 * The function reverses a given string by splitting it into an array of characters, reversing the
 * order of the array, and then joining the characters back into a string.
 * @param str - a string that needs to be reversed.
 * @returns The function `ReverseString` returns a reversed version of the input string `str`. It
 * achieves this by splitting the string into an array of characters, reversing the order of the array,
 * and then joining the characters back into a string.
 */
function ReverseString(str) {
    return str.split("").reverse().join("");

    /*
    var result= "";
    for(let i= str.length -1; i>=0; i--) {
        result +=str[i];
    }
    return result;
    */
}

console.log(ReverseString("Hello")); // return olleH

File: reverseString.cs
using System;

class MainClass
{
   /// <summary>
   /// The function reverses a given string by iterating through each character and concatenating it to
   /// the beginning of a new string.
   /// </summary>
   /// <param name="str">The input string that needs to be reversed.</param>
    public static string FirstReverse(string str)
    {
        string result = "";
        foreach (char c in str)
        {
            result = c + result;
        }
    }

    static void main() {
        Console.WriteLine(FirstReverse("Hello"));
    }
}
